hvillacrusisCT/dotTheory---TypeScript
# JS-706: Next Generation JavaScript and TypeScript

## Related Ticket
[JS-706](https://codeandtheory.atlassian.net/browse/JS-706)

## Description
This pull request introduces three new utility functions designed to showcase different features of TypeScript:

1. `arrowFunction`: This is an arrow function that takes a default parameter and any number of rest parameters. It then combines all of these into an array, which it returns.
   
2. `arrayFunction`: This function returns a hardcoded array with three elements.
   
3. `objectFunction`: This function returns a hardcoded object with three key-value pairs.

In addition, we have included variable declarations that destructure the results of `arrayFunction` and `objectFunction`, and several `console.log` statements to demonstrate the output of these functions.

Each function and variable has been properly typed to adhere to TypeScript's static typing system, ensuring safer and more predictable code.

## How to Test
To verify the changes introduced by this PR, you can follow these steps:

1. Start the `lite-server` in the project directory by running `lite-server`.
2. Open your web browser and navigate to the URL provided by `lite-server` (typically `http://localhost:3000`).
4. Open your browser's developer tools, then go to the Console tab.
7. Here you should see the console outputs of the following actions:

```markdown
- Test `arrowFunction` by calling it with a variety of arguments. For example, `arrowFunction('Custom', 'rest1', 'rest2', 'rest3')`.
- Test `arrayFunction` and `objectFunction` simply by calling them: `arrayFunction()` and `objectFunction()`.
- Check the console for the output of the destructuring declarations. You should see the elements of the array and the values of the object displayed in separate console.log statements.
